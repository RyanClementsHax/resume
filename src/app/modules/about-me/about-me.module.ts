import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { ParallaxModule } from 'ngx-parallax';

import { AboutMeComponent } from './containers/about-me/about-me.component';

@NgModule({
  imports: [
    CommonModule,
    ParallaxModule
  ],
  declarations: [
    AboutMeComponent
  ]
})
export class AboutMeModule {}
