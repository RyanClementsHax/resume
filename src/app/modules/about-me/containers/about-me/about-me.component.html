<header parallax [config]="parallaxConfig" class="jumbotron">
    <div class="title-container">
        <h1>Ryan&nbsp;Clements</h1>
        <h2><em>Chief&nbsp;Coffee Fetcher&nbsp;Extraordinaire</em></h2>
    </div>
</header>
<main>
    <article>
        <header class="content-header">
            <h2>About the Man, the Myth, the Legend</h2>
            <a
                href="https://www.youtube.com/watch?v=8F9jXYOH2c0"
                rel="noopener noreferrer"
                target="_blank">
                <i class="fas fa-external-link-alt"></i>
            </a>
        </header>
        <p>
            I love coding, good dogs, and looking cool in suits. I have had many amazing experiences being able 
            to create awesome software. From front end to back end, I seem to enjoy anything that can get me in front 
            of a computer to do what I love to do most: <em>break things until they work</em>. In my spare time I like 
            working out, going to church, and making cool stuff (like this website #shamelessPlug).
        </p>
    </article>
</main>