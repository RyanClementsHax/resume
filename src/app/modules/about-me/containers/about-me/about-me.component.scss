@import "variables.scss";

.jumbotron {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    height: $jumbotron-height;
    background-image: url('/assets/img/headshot-shades-portrait.jpg');
    background-size: auto 150%;

    .title-container {
        padding: $content-padding;
        margin: $content-padding;
        border-radius: $default-border-radius;
        box-shadow: 3px 3px 4px rgba(0,0,0,0.5);
        background: rgba(255, 255, 255, 0.8);

        h1 {
            margin: 0;
        }
    
        h2 {
            color: $color-accent;
            margin: 0;
        }
    }
    
    @media screen and (min-width: 680px) {
        background-image: url('/assets/img/headshot-on-phone.jpg');
        background-size: auto 125%;
    }
    
    @media screen and (min-width: 1280px) {
        background-size: 125%;
    }
}

.content-header {
    display: flex;
    align-items: center;
    margin: 20px 0;
    
    h2 {
        margin: 0;
    }

    a {
        margin-left: auto;

        &:visited {
            color: initial;
        }
    }
}

article {
    margin: auto;
    max-width: $content-width;
    padding: $content-padding;
}