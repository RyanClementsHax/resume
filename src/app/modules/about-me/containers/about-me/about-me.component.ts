import { Component } from '@angular/core';

import { ParallaxConfig } from 'ngx-parallax';

@Component({
  selector: 'about-me',
  templateUrl: './about-me.component.html',
  styleUrls: ['./about-me.component.scss']
})
export class AboutMeComponent {
  parallaxConfig: ParallaxConfig = {
    initialValue: 50,
    ratio: -0.2
  };
}
