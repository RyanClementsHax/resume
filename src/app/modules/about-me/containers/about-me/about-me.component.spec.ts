import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { AboutMeComponent } from './about-me.component';

describe('HomePageComponent', () => {
  let component: AboutMeComponent;
  let fixture: ComponentFixture<AboutMeComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AboutMeComponent],
      schemas: [NO_ERRORS_SCHEMA]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AboutMeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
