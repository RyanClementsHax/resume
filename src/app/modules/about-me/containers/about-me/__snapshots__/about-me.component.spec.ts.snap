// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageComponent should match snapshot 1`] = `
AboutMeComponent {
  "parallaxConfig": Object {
    "initialValue": 50,
    "ratio": -0.2,
  },
}
`;
